35_feature_main_activity

VSemikoz opened the pull request.

VSemikoz committed:
add main activity layout

VSemikoz committed:
add news item layout for RecView adapter

VSemikoz committed:
add color text into resources

VSemikoz committed:
config preview of news_item

arudometkin commented on app/src/main/res/layout/news_item.xml:
```diff
@@ -0,0 +1,60 @@
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent"
+        tools:src="@drawable/image_example" />
```
Not bad but still required to upload an image. I meant something like this: `tools:src="@tools:sample/backgrounds/scenic"`
More information [here](http://developer.alexanderklimov.ru/android/studio/tools.php). It will allow using pre-loaded resources instead of uploading your own. So, you'll be able to remove your image.

arudometkin commented on app/src/main/res/layout/news_item.xml:
```diff
@@ -0,0 +1,60 @@
+
+
+    <ImageView
+        android:id="@+id/imageView"
```
Please, use the same rule for naming. If you decided to use C-style notation, then continue using it across your layouts. Dont's switch from it to camel case and back. So, `imageView` will be `image_view`. If you want to keep camel case, then keep it and rename others, so `home_view` will become `homeView`. Please, make a decision on this. We usually recomend to use C-style for identifiers.

arudometkin commented on app/src/main/res/layout/news_item.xml:
```diff
@@ -0,0 +1,60 @@
+
+    <TextView
+        android:id="@+id/title"
+        android:layout_width="0dp"
```
All sizes and dimensions should reside in dimens. Please, refer to official [guide](https://developer.android.com/guide/topics/resources/more-resources.html#Dimension) and all related stuff to dimens.

arudometkin reviewed (changes requested):
LG. BTW, where's activity and fragment code?

VSemikoz committed:
config news_item preview using tools:samples

VSemikoz committed:
rename view's id in c-style

VSemikoz committed:
add news feed fragment

VSemikoz committed:
change layouts to fragment model:

 - change activity main layout to fragment carrier
 - add news_feed_fragment as fragment
 - inflate fragment by news_feed_fragment

VSemikoz committed:
news model TextView refactor

VSemikoz committed:
add newsFeedAdapter

VSemikoz committed:
Set NewsFeedAdapter in mainActivity

VSemikoz committed:
refactor views id in viewType_viewName style

VSemikoz committed:
refactor dimens of views
 - add dimens.xml with dimens of views
 - replace all dimens of views by values of dimens.xml

VSemikoz committed:
remove filter model from NewsFeedAdapter

VSemikoz committed:
add id for fragment

VSemikoz committed:
resize image on ImageView

VSemikoz committed:
add constructor for tests

VSemikoz committed:
init test data for RecView

VSemikoz committed:
remove top action bar

VSemikoz committed:
change color primary to white

VSemikoz committed:
add text title and desc size

VSemikoz committed:
set title and desc text size from dimens

arudometkin commented on app/src/main/java/ru/vssemikoz/newsfeed/adapters/NewsFeedAdapter.java:
```diff
@@ -0,0 +1,55 @@
+    @NonNull
+    @Override
+    public NewsFeedAdapter.NewsViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+        LayoutInflater inflater = LayoutInflater.from(parent.getContext());
```
Let's combine this and next line and remove redundant variables. We don't need a LayoutInflater as a variable here.

arudometkin commented on app/src/main/java/ru/vssemikoz/newsfeed/fragments/newsFeedFragment.java:
```diff
@@ -0,0 +1,30 @@
+    }
+
+    @Override
+    public void onPause() {
```
There's no need in overriding a method without any logic. Let's remove it in case it's not required.

arudometkin commented on app/src/main/java/ru/vssemikoz/newsfeed/fragments/newsFeedFragment.java:
```diff
@@ -0,0 +1,30 @@
+public class newsFeedFragment extends Fragment {
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
```
There's no need in overriding a method without any logic. Let's remove it in case it's not required.

arudometkin commented on app/src/main/java/ru/vssemikoz/newsfeed/fragments/newsFeedFragment.java:
```diff
@@ -0,0 +1,30 @@
+
+import ru.vssemikoz.newsfeed.R;
+
+public class newsFeedFragment extends Fragment {
```
A class name should start with the upper letter.

arudometkin commented on app/src/main/java/ru/vssemikoz/newsfeed/models/NewsItem.java:
```diff
@@ -68,6 +74,8 @@ public void setImageUrl(URL imageUrl) {
     public void setPublishedAt(Date publishedAt) {
         this.publishedAt = publishedAt;
     }
+
```
Please, try to avoid unnecessary empty lines. They're placed at the end of the file and between code parts. But they shouldn't follow 2 and more in a row.

arudometkin commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -1,18 +1,15 @@
-        app:layout_constraintRight_toRightOf="parent"
-        app:layout_constraintTop_toTopOf="parent" />
+    <fragment
+        android:id="@+id/fr_newsfeed"
```
I think we're talking about the news feed. So, the correct name would be `NewsFeed` for classes and `fr_news_feed` here.

arudometkin commented on app/src/main/res/layout/news_feed_fragment.xml:
```diff
@@ -0,0 +1,64 @@
+        app:srcCompat="@android:drawable/btn_star_big_on"/>
+
+    <ImageView
+        android:id="@+id/im_home"
```
Have you put `im` by error? Looks like the correct prefix is `iv`.

arudometkin commented on app/src/main/res/layout/news_item.xml:
```diff
@@ -0,0 +1,61 @@
+        tools:text="@tools:sample/lorem" />
+
+    <ImageButton
+        android:id="@+id/im_follow"
```
The same here, we're using `ImageButton`, so the abbreviation is `ib`.

arudometkin commented on app/src/main/res/values/dimens.xml:
```diff
@@ -0,0 +1,16 @@
+    <dimen name="category_height">60dp</dimen>
+    <dimen name="buttons_height">40dp</dimen>
+
+    <dimen name="text_size_title">24dp</dimen>
```
Please, revisit the dimens guide and pick up correct unit for text/font sizes. Texts and fonts are not mesured in `dp`.

arudometkin reviewed (approved):
LG%C

VSemikoz committed:
combine lines to avoid redundant variables

VSemikoz committed:
remove overriding methods without logic

VSemikoz committed:
rename class to up first latter

VSemikoz committed:
remove unnecessary empty lines

VSemikoz committed:
correct id's prefixes in layouts

VSemikoz committed:
correct measure units for text size
